capstone-backend/src/main/java/com/google/sps/data/Group.java

```diff
@@ -0,0 +1,117 @@
+package com.google.sps.data;
+
+import java.util.ArrayList;
+import java.util.List;
+
+ 
+public abstract class Group extends BaseEntity {
+  
+  private String name;
+  private String description;
+  private String announcement;
+  private List<String> posts;
+  private String ownerID;
+  private List<String> memberIDs;
+  private List<String> inviteIDs;
+
+  public Group(String name, String description, String announcement, List<String> posts, 
+      String ownerID, List<String> memberIDs, List<String> inviteIDs) {
+        this.name = name;
+        this.description = description;
+        this.announcement = announcement;
+        this.posts = posts;
+        this.ownerID = ownerID;
+        this.memberIDs = memberIDs;
+        this.inviteIDs = inviteIDs;
+  }
+
+  public Group() {
+    this("", "", "", new ArrayList<>(), "", new ArrayList<>(), new ArrayList<>());
+  }
+ 
+  public String getName() {
+    return this.name;
+  }
+ 
+  public void setName(String name) {
+    this.name = name;
+  }
+ 
+  public String getDescription() {
+    return this.description;
+  }
+ 
+  public void setDescription(String description) {
+    this.description = description;
+  }
+
+  public String getAnnouncement() {
+    return this.announcement;
+  }
+ 
+  public void setAnnouncement(String announcement) {
+    this.announcement = announcement;
+  }
+
+  public List<String> getPosts() {
+    return this.posts;
+  }
+
+  public void post(String post) {
```

Comment:
nit: I think `addPost` would be clearer here.

Reply:
I agree, I changed this!